src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -620,4 +620,51 @@ BOOST_AUTO_TEST_CASE(CustomLayerTest)
   arma::mat prediction = arma::zeros<arma::mat>(1, predictionTemp.n_cols);
 }
 
+/**
+ * Test the overload of Forward function which allows partial forward pass.
+ */
+BOOST_AUTO_TEST_CASE(PartialForwardTest)
+{
+  FFN<NegativeLogLikelihood<>, RandomInitialization> model;
+  model.Add<Linear<> >(5, 10);
+
+  // Add a new Add<> module which adds a constant term to the input.
+  Add<>* AddModule = new Add<>(10);
+  model.Add(AddModule);
+
+  LinearNoBias<>* LinearNoBiasModule = new LinearNoBias<>(10, 10);
+  model.Add(LinearNoBiasModule);
```

Comment:
Weird, the comment didn't seem to post.  I wonder what I did wrong.

Anyway: whenever you use `new`, you have to also use `delete` otherwise there is a memory leak.  So at the end of this test you'd need to add that.  But I would say, in this case, you can actually do something a little easier and avoid calling `new` at all:

```
Add<> addModule(10);
model.Add(addModule);
```

(Also note the lowercase variables, please do that to match the style guide. :))

Sorry this didn't post earlier!